process/notifier/sovereignNotifier_test.go

```diff
@@ -0,0 +1,254 @@
+package notifier
+
+import (
+	"encoding/hex"
+	"strings"
+	"testing"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-core-go/data/block"
+	"github.com/multiversx/mx-chain-core-go/data/outport"
+	"github.com/multiversx/mx-chain-core-go/data/transaction"
+	"github.com/multiversx/mx-chain-core-go/marshal"
+	"github.com/multiversx/mx-chain-sovereign-notifier-go/testscommon"
+	"github.com/stretchr/testify/require"
+)
+
+func createArgs() ArgsSovereignNotifier {
+	return ArgsSovereignNotifier{
+		Marshaller:          &testscommon.MarshallerMock{},
+		SubscribedAddresses: [][]byte{[]byte("address")},
+	}
+}
+
+func createBlockData(marshaller marshal.Marshalizer) *outport.BlockData {
+	headerV2 := &block.HeaderV2{
+		Header:            &block.Header{},
+		ScheduledRootHash: []byte("root hash"),
+	}
+
+	headerBytes, _ := marshaller.Marshal(headerV2)
+	return &outport.BlockData{
+		HeaderBytes: headerBytes,
+		HeaderType:  string(core.ShardHeaderV2),
+	}
+}
+
+func TestNewSovereignNotifier(t *testing.T) {
+	t.Parallel()
+
+	t.Run("should work", func(t *testing.T) {
```

Comment:
theses t.Runs can be parallelized as well

Reply:
There would be almost no improvement for this; these are just constructor tests